aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1991,6 +1989,12 @@ Tensor squeeze(const Tensor& self, int64_t dim) {
   return result;
 }
 
+Tensor squeeze_quantized(const Tensor& self, int64_t dim) {
```

Comment:
can we merge the two variants of squeeze_qtensor, and also merge the implementations of squeeze_quantized (call one squeeze_quantized from another, ideally we can call squeeze_quantized(const Tensor& self, int64_t dim) from squeeze_quantized(const Tensor& self))

Reply:
I should be able to merge the 2 squeeze_qtensor into a single function by making the dim arg an optional arg and then make some modifications to handle each case based on whether the optional arg is present or not. Is this what you mean, or do you mean I should call squeeze_qtensor(const Tensor& self, int64_t dim) from within the for loop of squeeze_qtensor(const Tensor& self)?

Re merging squeeze_quantized:
I'll follow up with you through work chat. There might be an issue with the previous un-dispatched version

Reply:
> I should be able to merge the 2 squeeze_qtensor into a single function by making the dim arg an optional arg and then make some modifications to handle each case based on whether the optional arg is present or not. Is this what you mean, or do you mean I should call squeeze_qtensor(const Tensor& self, int64_t dim) from within the for loop of squeeze_qtensor(const Tensor& self)?

oh I was assuming the no-dim variant is a special case for the dim-variant. if so we can do the first things you proposed here, if not, then I feel it might be OK to keep them this way. But looking at the implementation of squeeze, it looks like there can be a lot of code sharing happening between these two variants, wondering if we can still share the code between the two squeeze_qtensor variants (currently it's duplicated).

yeah modification for squeeze_quantized will depend on how we implement squeeze_qtensor.

Reply:
optional argument for dim: we can do this in implementation, not in interface/signature, we still want to exactly match the fp32 signature

Reply:
sorry I don't understand. are you suggesting we should make dim an optional argument in squeeze_quantized. if so, I'm confused how we can get this to work without modifying native_functions.yaml

Reply:
done. can you approve?